File: app/src/main/java/com/kinwatt/powermeter/ui/RecordListActivity.java
package com.kinwatt.powermeter.ui;

import android.content.Intent;
import android.support.design.widget.FloatingActionButton;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.kinwatt.powermeter.R;
import com.kinwatt.powermeter.data.Record;
import com.kinwatt.powermeter.data.provider.RecordProvider;
import com.kinwatt.powermeter.ui.fragments.RecordFragment;

public class RecordListActivity extends ActivityBase implements RecordFragment.OnListFragmentInteractionListener {

    private RecordFragment recordFragment;
    private TextView noDevices;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_record_list);

        noDevices = findViewById(R.id.no_records);
        recordFragment = (RecordFragment) getSupportFragmentManager().findFragmentById(R.id.records);

        FloatingActionButton fab = findViewById(R.id.add);
        fab.setOnClickListener(view -> {
            Intent intent = new Intent(this, MainActivity.class);
            startActivity(intent);
        });

        if (RecordProvider.getProvider(this).getAll().isEmpty()) {
            recordFragment.getView().setVisibility(View.GONE);
        }
        else {
            noDevices.setVisibility(View.GONE);
        }
    }

    @Override
    public void onListFragmentInteraction(Record item) {
        Intent intent = new Intent(this, RecordSummaryActivity.class);
        intent.putExtra("file_name", RecordProvider.getProvider(this).getFile(item).getAbsolutePath());
        startActivity(intent);
    }
}

File: app/src/main/java/com/kinwatt/powermeter/ui/activities/RecordListActivity.kt
package com.kinwatt.powermeter.ui.activities

import android.content.Intent
import com.google.android.material.floatingactionbutton.FloatingActionButton
import android.os.Bundle
import android.view.View

import com.kinwatt.powermeter.R
import com.kinwatt.powermeter.data.Record
import com.kinwatt.powermeter.data.provider.RecordProvider
import com.kinwatt.powermeter.databinding.ActivityRecordListBinding
import com.kinwatt.powermeter.ui.fragments.RecordFragment

class RecordListActivity : ActivityBase(), RecordFragment.OnListFragmentInteractionListener {

    private lateinit var binding: ActivityRecordListBinding
    private lateinit var recordFragment: RecordFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityRecordListBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        recordFragment = supportFragmentManager.findFragmentById(R.id.records) as RecordFragment

        val fab = findViewById<FloatingActionButton>(R.id.add)
        fab.setOnClickListener { view ->
            val intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
        }

        if (RecordProvider.getProvider(this).all.isEmpty()) {
            recordFragment.view!!.visibility = View.GONE
        } else {
            binding.noRecords.visibility = View.GONE
        }
    }

    override fun onListFragmentInteraction(item: Record) {
        val intent = Intent(this, RecordSummaryActivity::class.java)
        intent.putExtra("file_name", RecordProvider.getProvider(this).getFile(item).absolutePath)
        startActivity(intent)
    }
}
